config/clusters/2i2c/staging.values.yaml

```diff
@@ -1,9 +1,18 @@
+dex:
+  enabled: true
+  hubHostName: staging.2i2c.cloud
+
+staticSites:
+  enabled: true
+  repo: https://github.com/yuvipanda/test-repo-push
+  branch: master
+  host: staging.2i2c.cloud
+  path: /textbook
+  githubApp:
+    enabled: true
```

Comment:
I think an increased verbosity in the naming of these will be helpful. If you read the config names and feel very confident what they are about.

- `staticSites`
   For now I understand there is an assumption to only support exposing one static content website. That seems fine to me and support for more can be added over time with some refactoring. The name is currently `staticSites` in plural. What about `staticWebsite` instead? I think that helps readers of this config better understand what this is about, because `site` is a little vague in the context of reading config assuming you try to understand what what the config is about.
   
- `repo` and `branch` are related to a git repository to be synced. `host` and `path` is about Ingress resources and how to access them. `githubApp` was something I didn't understand what it was about until I read about it, but I now understand it as configuration of credentials to access the git repo to sync from.

  I propose more explicit names like `gitRepoUrl`, `gitRepoBranch`, `ingressHost`, `ingressPath`, and perhaps `gitRepoCredentials.enabled`, with nested values `githubAppId`, `githubAppKey`.
  
I've grown to value clearly named config as even more important than clearly named variables and functions, because config names are like variable names exposed to an even wider audience.

Reply:
@consideRatio thank you for this very helpful feedback. I've renamed and moved around config so it is more self descriptive. I have preferred nesting over compound words, as I think that expresses intent clearer in a hierarchical config format like YAML. I hope this is good from your pov.

Reply:
Thanks a lot for this thoughtful response, @consideRatio. You've convinced me easily it should be called `ingress` and not `url` and I've made that change.

Re nested vs flat, I very much prefer nested for the following reasons:

1. Avoids repetition in the name, making it easier to read the config.
2. Allows expansion in the future - we can add a `source.type` that can be set to `git` or `google-drive` or whatever, and we can add another `googleDrive` key at that point without breaking anything under `git`. This combination of setting `type` and then nesting keys under that is one of my fav k8s patterns.

Thank you for thinking deeply about this, and communicating it clearly!